### PR TITLE
feat: Add PoolUrl dataclass 

### PR DESCRIPTION
**Pull request by Ytemiloluwa**
This PR adds the poolUrl dataclass to pools.py which was suggested in PR #145 @b-rowan and here are the changes made: 

-  Created a `Scheme` Enum with `STRATUM_V1` and `STRATUM_V2` variants.
-  Defined the `PoolUrl` dataclass with `scheme`, `host`, `port`, and an optional `pubkey`.
-  Updated `PoolMetrics` to include a `PoolUrl` attribute.

I plan to modify the `_get_pools` methods to parse URLs and populate `poolUrl` instances.

### CODE REVIEW

File: pyasic/data/pools.py [hunk 1]
```diff
@@ -19,15 +39,15 @@ class PoolMetrics:
     pool_stale_percent: Percentage of stale shares by the pool.
     """
 
-    accepted: int = None
```
Comment (b-rowan) [inline pyasic/data/pools.py:22]: Probably want to keep these defaulting to none.

**Review by b-rowan — CHANGES_REQUESTED** [review]

File: pyasic/data/pools.py (hunk 1 above)
Comment (Ytemiloluwa) [reply pyasic/data/pools.py:22]: Okay

File: pyasic/miners/backends/braiins_os.py [hunk 2]
```diff
@@ -720,6 +720,10 @@ async def upgrade_firmware(self, file: Path):
             "_get_uptime",
             [RPCAPICommand("rpc_summary", "summary")],
         ),
+        str(DataOptions.POOLS): DataFunction(
```
Comment (b-rowan) [inline pyasic/miners/backends/braiins_os.py:723]: Can you remove the get_pools implementation for bosminer from this?  Want to keep this PR atomic as much as possible.  Any other get_pools implementatikons already done should also be updated to use the new PoolUrl dataclass.

File: pyasic/data/pools.py [hunk 3]
```diff
@@ -1,4 +1,24 @@
...
+    scheme: Scheme
+    host: str
+    port: int
+    pubkey: Optional[str] = None
+
+    def __str__(self) -> str:
+        if self.scheme == Scheme.STRATUM_V2 and self.pubkey:
+            return f"{self.scheme.value}://{self.host}:{self.port}/{self.pubkey}"
+        else:
+            return f"{self.scheme.value}://{self.host}:{self.port}"
```
Comment (b-rowan) [inline pyasic/data/pools.py:21]: Maybe want a `from_str` function that parses a string formatted pool URl into this format.  You could likely use something from stdlib like urllib.

**Review by b-rowan — CHANGES_REQUESTED** [review]

File: pyasic/miners/backends/braiins_os.py (hunk 2 above)
Comment (Ytemiloluwa) [reply pyasic/miners/backends/braiins_os.py]: I have rebased and dropped the unwanted commits.

File: pyasic/data/pools.py (hunk 3 above)
Comment (Ytemiloluwa) [reply pyasic/data/pools.py:21]: do you mean I should replace ` __str__(self)` with `from_str`?

Comment (Ytemiloluwa) [reply pyasic/data/pools.py:21]: which format should be the desired output? 

1. stratum2+tcp://btc.pool.com:3333/pubkey
2. PoolUrl(scheme=<Scheme.STRATUM_V2: 'stratum2+tcp'>, host='btc.pool.com', port=3333, pubkey='pubkey')

Comment (b-rowan) [reply pyasic/data/pools.py:21]: Precisely.

Comment (Ytemiloluwa) [reply pyasic/data/pools.py:21]: okay, I will push an update soon.

File: pyasic/data/pools.py [hunk 4]
```diff
@@ -19,17 +50,17 @@ class PoolMetrics:
...
     accepted: int = None
     rejected: int = None
     get_failures: int = None
     remote_failures: int = None
     active: bool = None
     alive: bool = None
-    url: str = None
     index: int = None
     user: str = None
-    pool_rejected_percent: float = field(init=False)
```
Comment (b-rowan) [inline pyasic/data/pools.py:31]: These dont need to be removed?

**Review by b-rowan — CHANGES_REQUESTED** [review]

File: pyasic/data/pools.py (hunk 4 above)
Comment (Ytemiloluwa) [reply pyasic/data/pools.py:31]: I dont understand what you mean. can you give more clarity please.

**Review by b-rowan — APPROVED** [review]

**Pull request merged**